### PR TITLE
Remove deprecations in preparation for 1.0

### PR DESCRIPTION
**Pull request by segiddins**
See https://github.com/CocoaPods/CocoaPods/issues/3632.

### CODE REVIEW

File: lib/cocoapods-core/version.rb [hunk 1]
```diff
@@ -46,14 +46,11 @@ class Version < Pod::Vendor::Gem::Version
...
     #         A string representing a version, or another version.
     #
-    # @todo   The `from` part of the regular expression should be remove in
-    #         CocoaPods 1.0.0.
-    #
     def initialize(version)
       if version.is_a?(Version) && version.head?
         version = version.version
         @head = true
-      elsif version.is_a?(String) && version =~ /HEAD (based on|from) (.*)/
```
Comment (kylef) [inline lib/cocoapods-core/version.rb]: Do we know what version `from` was used in?

**Comment by kylef** [comment]
:ok_hand:

File: lib/cocoapods-core/version.rb (hunk 1 above)
Comment (segiddins) [inline lib/cocoapods-core/version.rb]: A long long time ago

-Samuel E. Giddins

On Jun 6, 2015, at 6:14 PM, Kyle Fuller <notifications@github.com> wrote:

In lib/cocoapods-core/version.rb:

>      def initialize(version)
>        if version.is_a?(Version) && version.head?
>          version = version.version
>          @head = true
> -      elsif version.is_a?(String) && version =~ /HEAD (based on|from) (.*)/
Do we know what version from was used in?

Comment (segiddins) [inline lib/cocoapods-core/version.rb]: (aka more than two years ago)

**Pull request closed without merging**